cmake/modules/CheckCompiler.cmake

```diff
@@ -122,7 +122,7 @@ if(cxx14)
   endif()
 endif()
 if(cxx17)
-  CHECK_CXX_COMPILER_FLAG("-std=c++1z" HAS_CXX17)
+  CHECK_CXX_COMPILER_FLAG("-std=c++17" HAS_CXX17)
```

Comment:
I would prefer to set `CMAKE_CXX_STANDARD` and let CMake decide what flag to add. Can I do that instead?

Reply:
Can we do that right after? I'd not like to change the behavior this fundamentally in ROOT 6.16, and I'd like to start from the 6.16 behavior for your (very, *very* welcome!) `CMAKE_CXX_STANDARD` modification.

Reply:
Ok, no problem. I can do that only on master then after you merge this.